### PR TITLE
[ISSUE 3407] sub task1 add the logic of annotation on the splicing class for apache dubbo client

### LINKED ISSUE #3407: [Task] add the logic of annotation on the splicing class for all rpc shenyu client
### Description

As a subtask of https://github.com/apache/incubator-shenyu/issues/3317, all rpc shenyu client needs to add the logic of annotation on the splicing class, you can see the `shenyu-client-springmvc` module.

### Task List

1. - [x] add the logic of annotation on the splicing class for apache dubbo client, @renzhuyan ;
2. - [ ] add the logic of annotation on the splicing class for alibaba dubbo client, @sunshujie1990 ;
3. - [ ] add the logic of annotation on the splicing class for grpc client, @hutaishi ;
4. - [ ] add the logic of annotation on the splicing class for motan client, @ChineseTony ;
5. - [ ] add the logic of annotation on the splicing class for sofa client, @ShawnSiao ;
6. - [x] add the logic of annotation on the splicing class for tars client, @gaochuang98 ;

### PR DESCRIPTION
**Pull request by renzhuyan**
// Describe your PR here; eg. Fixes #3407

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java [hunk 1]
```diff
@@ -118,18 +125,47 @@ private void handler(final ServiceBean<?> serviceBean) {
...
-                publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuDubboClient, method));
+            ShenyuDubboClient methodShenyuClient = method.getAnnotation(ShenyuDubboClient.class);
+            if (Objects.nonNull(methodShenyuClient)) {
+                publisher.publishEvent(buildMetaDataDTO(serviceBean, methodShenyuClient, method, superPath));
+            }
+        }
+    }
+
+    private String buildApiSuperPath(@NonNull final Class<?> clazz) {
+        ShenyuDubboClient shenyuDubboClient = AnnotationUtils.findAnnotation(clazz, ShenyuDubboClient.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java:148]: Why call `AnnotationUtils.findAnnotation(clazz, ShenyuDubboClient.class);` twice ?

**Comment by dragon-zhang** [comment]
Check `shenyu-integrated-test-apache-dubbo` module to add more test code, suggest you read https://shenyu.apache.org/blog/IntegrationTest-Analysis/ before continuing development

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java (hunk 1 above)
Comment (renzhuyan) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java:148]: Once to get the path and once to build metadatadto

Comment (dragon-zhang) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java:148]: Can we get it only once?

Comment (renzhuyan) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java]: Of course, I have modified it

**Review by dragon-zhang — APPROVED** [review]

**Pull request merged**